Lib/importlib/_bootstrap_external.py

```diff
@@ -1470,6 +1470,9 @@ def invalidate_caches():
         # https://bugs.python.org/issue45703
         _NamespacePath._epoch += 1
 
+        from importlib.metadata import MetadataPathFinder
+        MetadataPathFinder().invalidate_caches()
```

Comment:
@brettcannon I included you in the review for this section. Any concerns with adding this hook here?